satpy/tests/compositor_tests/test_spectral.py

```diff
@@ -83,34 +85,50 @@ class TestNdviHybridGreenCompositor:
+            dims=("y", "x"), attrs={"name": "C04"})
 
     def test_ndvi_hybrid_green(self):
         """Test General functionality with linear scaling from ndvi to blend fraction."""
-        comp = NDVIHybridGreen("ndvi_hybrid_green", limits=(0.15, 0.05), prerequisites=(0.51, 0.65, 0.85),
-                               standard_name="toa_bidirectional_reflectance")
-
-        # Test General functionality with linear strength (=1.0)
-        res = comp((self.c01, self.c02, self.c03))
-        assert isinstance(res, xr.DataArray)
-        assert isinstance(res.data, da.Array)
-        assert res.attrs["name"] == "ndvi_hybrid_green"
-        assert res.attrs["standard_name"] == "toa_bidirectional_reflectance"
-        data = res.values
+        with dask.config.set(scheduler=CustomScheduler(max_computes=1)):
+            comp = NDVIHybridGreen("ndvi_hybrid_green", limits=(0.15, 0.05), prerequisites=(0.51, 0.65, 0.85),
+                                   standard_name="toa_bidirectional_reflectance")
+
+            # Test General functionality with linear strength (=1.0)
+            res = comp((self.c01, self.c02, self.c03))
+            assert isinstance(res, xr.DataArray)
+            assert isinstance(res.data, da.Array)
+            assert res.attrs["name"] == "ndvi_hybrid_green"
+            assert res.attrs["standard_name"] == "toa_bidirectional_reflectance"
+            data = res.values
         np.testing.assert_array_almost_equal(data, np.array([[0.2633, 0.3071], [0.2115, 0.3420]]), decimal=4)
 
+    def test_ndvi_hybrid_green_dtype(self):
+        """Test that the datatype is not altered by the compositor."""
+        with dask.config.set(scheduler=CustomScheduler(max_computes=1)):
+            comp = NDVIHybridGreen("ndvi_hybrid_green", limits=(0.15, 0.05), prerequisites=(0.51, 0.65, 0.85),
+                                   standard_name="toa_bidirectional_reflectance")
+
+            # Test General functionality with linear strength (=1.0)
+            res = comp((self.c01, self.c02, self.c03)).compute()
+        assert res.data.dtype == np.float32
+
     def test_nonliniear_scaling(self):
         """Test non-linear scaling using `strength` term."""
-        comp = NDVIHybridGreen("ndvi_hybrid_green", limits=(0.15, 0.05), strength=2.0, prerequisites=(0.51, 0.65, 0.85),
-                               standard_name="toa_bidirectional_reflectance")
+        with dask.config.set(scheduler=CustomScheduler(max_computes=1)):
+            comp = NDVIHybridGreen("ndvi_hybrid_green", limits=(0.15, 0.05), strength=2.0,
+                                   prerequisites=(0.51, 0.65, 0.85),
+                                   standard_name="toa_bidirectional_reflectance")
 
-        res = comp((self.c01, self.c02, self.c03))
-        np.testing.assert_array_almost_equal(res.values, np.array([[0.2646, 0.3075], [0.2120, 0.3471]]), decimal=4)
+            res = comp((self.c01, self.c02, self.c03)).compute()
```

Comment:
I don't think you need a separate test for dtype checks. I meant adding the checks to the existing tests. They are all checking the validity of the result so I think it still fits in the existing tests.

```suggestion
            res = comp((self.c01, self.c02, self.c03))
            res_np = res.data.compute()
            assert res.dtype == res_np.dtype
            assert res.dtype == np.float32
```